optuna/distributions.py

```diff
@@ -548,16 +548,9 @@ def json_to_distribution(json_str: str) -> BaseDistribution:
                 return FloatDistribution(low, high, log=log, step=step)
 
             else:
-                if log:
-                    if step is not None:
-                        return IntLogUniformDistribution(low, high, step)
-                    else:
-                        return IntLogUniformDistribution(low, high)
-                else:
-                    if step is not None:
-                        return IntUniformDistribution(low, high, step)
-                    else:
-                        return IntUniformDistribution(low, high)
+                if step is None:
+                    step = 1
+                return IntDistribution(low=low, high=high, log=log, step=step)
```

Comment:
[note] `json_to_distribution` creates `IntDistribution` for abbreviated format after this PR.